weed/shell/replication_health_checker.go

```diff
@@ -0,0 +1,586 @@
+package shell
+
+import (
+	"context"
+	"fmt"
+	"github.com/chrislusf/seaweedfs/weed/glog"
+	"github.com/chrislusf/seaweedfs/weed/operation"
+	"github.com/chrislusf/seaweedfs/weed/pb/master_pb"
+	"github.com/chrislusf/seaweedfs/weed/pb/volume_server_pb"
+	"github.com/chrislusf/seaweedfs/weed/storage"
+	"github.com/chrislusf/seaweedfs/weed/storage/needle"
+	"google.golang.org/grpc"
+	"io"
+	"math/rand"
+	"sort"
+	"strings"
+	"sync"
+)
+
+func init() {
+	commands = append(commands, &commandReplicationHealthChecker{})
+	commands = append(commands, &commandReplicationHealthRepair{})
+}
+
+type commandReplicationHealthChecker struct {
+	args       []string
+	commandEnv *commandEnv
+	writer     io.Writer
+	checker    *ReplicationHealthChecker
+}
+
+func (c *commandReplicationHealthChecker) Name() string {
+	return "volume.check.replication"
+}
+
+func (c *commandReplicationHealthChecker) Help() string {
+	return `check all replications of a volume, and find the volumes whose content is different in replications
+
+	This command-line will check all volumes in the topology, and find all unhealthy volumes whose content
+	are inconsistent in replications, finally, output the error volume IDs.
+	It will compare the file size, file count(the count of modify and update requests with the volume from
```

Comment:
fileCount minus deletedFileCount will get count for actual files. this value should be consistent for all volumes.